.github/workflows/ci-pipeline.yml

```diff
@@ -46,3 +46,52 @@ jobs:
 
       - name: Deploy to Azure Web App
         run: curl -dH -X POST ${{ secrets.AZURE_WEB_HOOK }}
+
+  azure-deploy:
+    name: deploy to Azure Web App
+    runs-on: ubuntu-latest
+    needs: docker-publish
+    steps:
+      - name: Deploy to Azure Web App
+        run: curl -dH -X POST ${{ secrets.AZURE_WEB_HOOK }}
+
+
+
+  terraform-deploy:
+    name: deploy using terraform
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged
+
+    defaults:
+      run:
+        working-directory: ./terraform
+
+    env:
+      ARM_ACCESS_KEY: ${{ secrets.ARM_ACCESS_KEY }}
+      ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+      ARM_CLIENT_SECRET: ${{ secrets.ARM_CLIENT_SECRET }}
+      ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+      ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+      TF_VAR_OAUTH_CLIENT_ID: ${{ secrets.OAUTH_CLIENT_ID }}
+      TF_VAR_OAUTH_CLIENT_SECRET: ${{ secrets.OAUTH_CLIENT_SECRET }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_wrapper: false
+
+      - run: terraform fmt
+      - run: terraform init
+
+      - run: terraform plan -no-color
```

Comment:
Similarly to the above, this plan isn't worth much if you just immediately apply underneath, unless you expect a human to be watching and to interrupt between these steps (and if so that feels like a poor solution to that scenario!)